portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderLocalServiceImpl.java

```diff
@@ -853,86 +854,44 @@ public DLFolder updateFolder(
 
 			// File entry types
 
-			DLFolder dlFolder = null;
+			DLFolder dlFolder = dlFolderLocalService.getDLFolder(folderId);
 
 			Set<Long> originalFileEntryTypeIds = new HashSet<>();
 
 			if (folderId > DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
 				originalFileEntryTypeIds = getFileEntryTypeIds(
 					dlFolderPersistence.getDLFileEntryTypes(folderId));
 
-				dlFolder = dlFolderLocalService.updateFolderAndFileEntryTypes(
-					serviceContext.getUserId(), folderId, parentFolderId, name,
-					description, defaultFileEntryTypeId, fileEntryTypeIds,
-					restrictionType, serviceContext);
-
-				if (!ExportImportThreadLocal.isImportInProcess()) {
-					_dlFileEntryTypeLocalService.cascadeFileEntryTypes(
-						serviceContext.getUserId(), dlFolder);
+				if (GetterUtil.getBoolean(
+						serviceContext.getAttribute(
+							"updateFolderAndFileEntryTypes"),
+						true)) {
+
+					dlFolder =
+						dlFolderLocalService.updateFolderAndFileEntryTypes(
+							serviceContext.getUserId(), folderId,
+							parentFolderId, name, description,
+							defaultFileEntryTypeId, fileEntryTypeIds,
+							restrictionType, serviceContext);
+
+					if (!ExportImportThreadLocal.isImportInProcess()) {
+						_dlFileEntryTypeLocalService.cascadeFileEntryTypes(
+							serviceContext.getUserId(), dlFolder);
+					}
 				}
-			}
-
-			// Workflow definitions
-
-			List<ObjectValuePair<Long, String>> workflowDefinitionOVPs =
-				new ArrayList<>();
-
-			if (restrictionType ==
-					DLFolderConstants.
-						RESTRICTION_TYPE_FILE_ENTRY_TYPES_AND_WORKFLOW) {
-
-				workflowDefinitionOVPs.add(
-					new ObjectValuePair<Long, String>(
-						DLFileEntryTypeConstants.FILE_ENTRY_TYPE_ID_ALL,
-						StringPool.BLANK));
-
-				for (long fileEntryTypeId : fileEntryTypeIds) {
-					String workflowDefinition = ParamUtil.getString(
-						serviceContext, "workflowDefinition" + fileEntryTypeId);
-
-					workflowDefinitionOVPs.add(
-						new ObjectValuePair<Long, String>(
-							fileEntryTypeId, workflowDefinition));
+				else {
+					dlFolder =
+						dlFolderLocalService.updateFolderAndFileEntryTypes(
```

Comment:
is it really needed to updateFolderAndFileEntryTypes in case of we don not have the paremeter updateFolderAndFileEntryTypes to true?

Reply:
yes, it's sadly needed @Mikellorza, based on what we get from the view